app/src/main/java/com/startup/go4lunch/MainActivity.java

```diff
@@ -253,12 +255,15 @@ private void updateHeaderLayoutWithUserData() {
     @Override
     public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         int id = item.getItemId();
+        final int YOUR_LUNCH = R.id.activity_main_drawer_your_lunch;
+        final int SETTING = R.id.activity_main_drawer_settings;
+        final int LOGOUT = R.id.activity_main_drawer_logout;
```

Comment:
Les constantes sont habituellement défini en champs de classe en tant que variables `private static` (petite optimisation)

Reply:
Aussi pour le naming, en les sortant de la méthode, il serait bien de donner plus de contexte, par exemple `YOUR_LUNCH_ITEM_ID`

Reply:
Ok Merci

Reply:
L'intérêt est que la variable ne sera pas recrée à chaque fois que l'on passe dans la méthode. Elle sera assignée en mémoire qu'une seule fois pour toutes les instances de MainActivity (si static).

Si tu veux la laisser dans la méthode, je te conseillerais juste de nommer les variables en minuscule. Par convention, une variable en majuscule est constante pour la classe entière.

Reply:
Ok

Reply:
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html
"The names of variables declared **class constants**"